frontend/src/pages/TermPlanner/ContextMenu/ContextMenu.jsx

```diff
@@ -41,21 +45,92 @@ const ContextMenu = ({ code, plannedFor }) => {
     dispatch(addTab(code));
   };
 
+  const [isEditMarkVisible, setIsEditMarkVisible] = useState(false);
+
+  const showEditMark = () => {
+    setIsEditMarkVisible(true);
+  };
+
+  const validLetterGrades = ["SY", "PS", "CR", "DN", "HD"];
+  const [markInputBuf, setMarkInputBuf] = useState(
+    useSelector((state) => state.planner.courses[code].mark),
+  );
+
+  const handleConfirmEditMark = (mark) => {
+    const attemptedMark = (` ${mark}`).slice(1).replace(" ", "");
+    if (
+      (attemptedMark.isNaN && !validLetterGrades.includes(attemptedMark))
+      || (parseFloat(attemptedMark) < 0 || parseFloat(attemptedMark) > 100)
+    ) {
+      return message.error("Could not update mark. Please enter a valid mark or letter grade");
+    }
+    dispatch(updateCourseMark({
+      code,
+      mark: attemptedMark,
+    }));
+    setIsEditMarkVisible(false);
+
+    validateTermPlanner(
+      dispatch,
+      { years, startYear, completedTerm },
+      { programCode, majors, minor },
+    );
+
+    return message.success("Mark Updated");
+  };
+
+  const [markUpdatedQueued, setMarkUpdateQueued] = useState(false);
+  useEffect(() => {
+    if (markUpdatedQueued) {
+      setMarkUpdateQueued(false);
+      handleConfirmEditMark(markInputBuf);
+    }
+  });
+
+  const queueEditMark = () => {
+    setMarkUpdateQueued(true);
+  };
+
+  const handleCancelEditMark = () => {
+    setIsEditMarkVisible(false);
+    setMarkUpdateQueued(false);
+  };
+
   return (
-    <Menu id={id} theme={theme.dark}>
-      {plannedFor && (
-        <Item onClick={handleUnschedule}>
-          <FaRegCalendarTimes className="context-menu-icon" /> Unschedule
+    <>
+      <Menu id={id} theme={theme.dark}>
+        {plannedFor && (
+          <Item onClick={handleUnschedule}>
+            <FaRegCalendarTimes className="context-menu-icon" /> Unschedule
+          </Item>
+        )}
+        <Item onClick={handleDelete}>
+          <DeleteFilled className="context-menu-icon" /> Delete from Planner
+        </Item>
+        <Item onClick={showEditMark}>
+          <FaEdit className="contextMenuIcon" /> Edit mark
+        </Item>
+        <Item onClick={handleInfo}>
+          <InfoCircleFilled className="context-menu-icon" />
+          View Info
         </Item>
-      )}
-      <Item onClick={handleDelete}>
-        <DeleteFilled className="context-menu-icon" /> Delete from Planner
-      </Item>
-      <Item onClick={handleInfo}>
-        <InfoCircleFilled className="context-menu-icon" />
-        View Info
-      </Item>
-    </Menu>
+      </Menu>
+      <Modal
```

Comment:
Do you mean a hierarchy that looks like
`ContextMenu` -> `EditMarksModal`

or 

`ContextMenu` -> `EditMarksWrapper` -> `EditMarks`, where `EditMarksWrapper` contains all the logic for mark editing.

The former would require us to build our own functionality for a `Modal` component (or at least the footer with the confirm/cancel buttons), which I don't take much issue with. Though it would require a bit of `css`, it may be preferable to the latter because it cuts out a need for another wrapper around `EditMarks`.

The only reason so much of the functionality exists within `ContextMenu` rather than `EditMarks` is due by default, the functions for edit and closing of the `Modal` must be passed into the `Modal` itself.

I have no issue with either, just lmk which is preferable.

Reply:
See comment below. I think we can just declare the `Modal` component and it's associated functions inside `EditMarks`? :)